compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmTypeGen.java

```diff
@@ -1433,7 +1434,7 @@ private static void loadTableType(MethodVisitor mv, BTableType bType) {
                 mv.visitInsn(DUP);
                 mv.visitLdcInsn((long) i);
                 mv.visitInsn(L2I);
-                mv.visitLdcInsn(fieldName);
+                mv.visitLdcInsn(StringEscapeUtils.unescapeJava(fieldName));
```

Comment:
Why the java escape? Not identifier encode?

Reply:
This field name is used to refer the `record` field name of the table key.  The record fields are mapped using the decoded field names. [escaped names == decode(encoded identifiers) ]
Eg: 
```ballerina
table key('emp\-id)

//field name passed as "emp\-id" at code gen (instruction-level)
//map key needed for record field access is "emp-id"
```